concrete/controllers/single_page/dashboard/system/environment/security.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Concrete\Controller\SinglePage\Dashboard\System\Environment;
+
+use Concrete\Core\Config\Repository\Repository;
+use Concrete\Core\Page\Controller\DashboardPageController;
+
+class Security extends DashboardPageController
+{
+    public function view()
+    {
+        /** @var Repository $config */
+        $config = $this->app->make(Repository::class);
+        $content_security_policy = (array) $config->get('concrete.security.misc.content_security_policy');
+        $this->set('content_security_policy', implode(PHP_EOL, $content_security_policy));
+        $this->set('strict_transport_security', $config->get('concrete.security.misc.strict_transport_security'));
+        $this->set('x_frame_options', $config->get('concrete.security.misc.x_frame_options'));
+    }
+
+    public function submit()
+    {
+        if (!$this->token->validate('update_security_policy')) {
+            $this->error->add($this->token->getErrorMessage());
+        }
+
+        if (!$this->error->has()) {
+            $content_security_policy = array_map('trim', explode(PHP_EOL, $this->post('content_security_policy')));
```

Comment:
What about using this?
```php
$content_security_policy = preg_split('/\s+/ms', (string) $this->post('content_security_policy'), -1, PREG_SPLIT_NO_EMPTY);
```
That way you don't assume that the received data contains the EOL of the server (and you don't have to call `trim` and check for empty strings).

Reply:
@mlocati 

This regex rule doesn't work for this case. CSP policies can include empty string like `default-src 'self'; frame-ancestors 'self'; form-action 'self';`

Reply:
So: what about this?

```php
$content_security_policy = preg_split('/[\r\n]+/ms', (string) $this->post('content_security_policy'), -1, PREG_SPLIT_NO_EMPTY);
```

Reply:
The  problem is: if you use a Windows server PHP_EOL will be "\r\n", and the explode function won't work if it receives a string that's using \n alone.

Reply:
So, generally speaking, I'd NEVER use PHP_EOL (also a few lines above when you send the data to the client: I'd still use "\n")

Reply:
> So: what about this?
> 
> ```
> $content_security_policy = preg_split('/[\r\n]+/ms', (string) $this->post('content_security_policy'), -1, PREG_SPLIT_NO_EMPTY);
> ```

This works as expected.